pylops_mpi/DistributedArray.py

```diff
@@ -1,12 +1,39 @@
-import numpy as np
-from typing import Optional, Union, Tuple, List
-from numbers import Integral
-from mpi4py import MPI
 from enum import Enum
+from numbers import Integral
+from typing import List, Optional, Tuple, Union
 
+import cupy as cp
+import cupy.cuda.nccl as nccl
+import numpy as np
+from mpi4py import MPI
 from pylops.utils import DTypeLike, NDArray
 from pylops.utils._internal import _value_or_sized_to_tuple
-from pylops.utils.backend import get_module, get_array_module, get_module_name
+from pylops.utils.backend import get_array_module, get_module, get_module_name
+
+cupy_to_nccl_dtype = {
```

Comment:
Similar to the above, I would suggest adding maybe a `nccl` file in `utils` where we add all NCCL related functions (or dicts like this one)...